### PR TITLE
Add Docker support and image CI

### PR DESCRIPTION
**Pull request by josecols**
This PR adds a Dockerfile and a GitHub Workflow/Action to build and publish it to DockerHub. The container has the following features:

- Support for `arm64` and `amd64`. The ARM support is important because docker emulation for `amd64` is slow, and one of the main issues reported in our cohort was getting MeTA to work on M1 chips.
- The binaries are made available globally, so a user can mount a volume and just run commands against their `config.toml`. So, tasks like profiling or PoS tagging just require the user to set up the `config.toml` and run the container. 
- MeTA is installed as a library in `/usr/local/lib` and `/usr/local/include`, so users can reference it in their C++ applications.
- The Readme file is updated with examples and simplified instructions.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+RUN apt-get install -y git cmake libjemalloc-dev zlib1g-dev
+RUN apt-get install -y clang-14 lld-14 libc++-14-dev libc++abi-14-dev
+
+WORKDIR /meta
+RUN git submodule update --init --recursive
+RUN mkdir build
+RUN sed -i 's:\.\./:/meta/:g' config.toml
+
+WORKDIR /meta/build
+RUN cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=On -DBUILD_STATIC_ICU=On -DLIBCXX_LIBRARY=/lib/llvm-14/lib/libc++.so.1
```
Comment (dmcguire81) [inline Dockerfile:20]: Nice.

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Docker Image CI
```
Comment (dmcguire81) [inline .github/workflows/main.yaml:1]: Looks great. I think we have a lot of options for wrapping this up.

**Review by dmcguire81 — APPROVED** [review]
Ship it!

File: .github/workflows/main.yaml (hunk 2 above)
Comment (josecols) [reply .github/workflows/main.yaml:1]: Thank you. After the initial release is ready, we can limit this action to run only on the `master` branch.

**Pull request merged**